src/ViewHelpers/JoinViewHelper.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file belongs to the package "TYPO3 Fluid".
+ * See LICENSE.txt that was shipped with this package.
+ */
+
+namespace TYPO3Fluid\Fluid\ViewHelpers;
+
+use TYPO3Fluid\Fluid\Core\Rendering\RenderingContextInterface;
+use TYPO3Fluid\Fluid\Core\ViewHelper\AbstractViewHelper;
+use TYPO3Fluid\Fluid\Core\ViewHelper\Traits\CompileWithRenderStatic;
+
+/**
+ * The JoinViewHelper combines elements from an array into a single string.
+ * You can specify both a general separator and a special one for the last
+ * element, which serves as the delimiter between the elements.
+ *
+ *
+ * Examples
+ * ========
+ *
+ * Simple join
+ * -----------
+ * ::
+ *
+ *    <f:join value="{0: '1', 1: '2', 2: '3'}" />
+ *
+ * .. code-block:: text
+ *
+ *    123
+ *
+ *
+ * Join with separator
+ * -------------------
+ *
+ * ::
+ *
+ *    <f:join value="{0: '1', 1: '2', 2: '3'}" separator=", " />
+ *
+ * .. code-block:: text
+ *
+ *    1, 2, 3
+ *
+ *
+ * Join with separator, and special one for the last
+ * -------------------------------------------------
+ *
+ * ::
+ *
+ *    <f:join value="{0: '1', 1: '2', 2: '3'}" separator=", " separatorLast=" and " />
+ *
+ * .. code-block:: text
+ *
+ *    1, 2 and 3
+ */
+final class JoinViewHelper extends AbstractViewHelper
+{
+    use CompileWithRenderStatic;
+
+    public function initializeArguments(): void
+    {
+        $this->registerArgument('value', 'array', 'An array', false);
+        $this->registerArgument('separator', 'string', 'The separator', false, '');
+        $this->registerArgument('separatorLast', 'string', 'The separator for the last pair.', false, null);
+    }
+
+    /**
+     * @param array $arguments
+     * @param \Closure $renderChildrenClosure
+     * @param RenderingContextInterface $renderingContext
+     *
+     * @return string The concatenated string
+     */
+    public static function renderStatic(array $arguments, \Closure $renderChildrenClosure, RenderingContextInterface $renderingContext)
+    {
+        $value = $arguments['value'] ?? null;
+        $separator = $arguments['separator'] ?? '';
+        $separatorLast = $arguments['separatorLast'] ?? null;
+
+        if ($value === null) {
+            $content = $renderChildrenClosure();
+            if (!is_array($content) && !$content instanceof \ArrayAccess && !$content instanceof \Traversable) {
```

Comment:
```suggestion
            if (!is_iterable($content)) {
```

We don't need `ArrayAccess` here.

Reply:
@mbrodala What do you mean with not needed ?

If that check is removed, object of \ArrayAccess throwing the exception instead of getting converted to an array later. This check is to prevent that. 

In case a object implements \ArrayAccess but not the \Traversable inteface (directly or indirectly) it would end in the exception with your suggestion, otherwise still hitting the conversion.

I'd say this should be then following:

```php
if (!is_iterable($content) && !$content instanceof \ArrayAccess)) {}
```

Reply:
Maybe we should cover the different cases with more tests to be sure everything works? I'm working on another VH and can create a basis for this (implementations of the different interfaces), so we can add more tests here as well.

Reply:
Once #863 is merged, I will adjust this PR accordingly.